### PR TITLE
Enforce platform-appropriate imports

### PR DESCRIPTION
**Pull request by Ryanseanlee**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by TomTirapani** [comment]
Looks great!

A minor (but important) thing, but could we ensure the new `xxx/register` imports are the first import, at the top of the list? It'll will help devs know at a glance if its always in the same place.

**Comment by Ryanseanlee** [comment]
So we found two cases where the mobile app was using desktop components. The two files are Chart.js and DockViewModel.js. 
For chart.js it is importing the useContextMenu and Anselm said @cnrudd you might have some insight on this. 
For DockViewModel.js it is also using ModelSupport @ghsolomon Anselm said this was recently added and if we could consider moving the entire DockView component into the desktop package.

**Comment by Ryanseanlee** [comment]
While we were troubleshooting this we changed the register file to throw so that we get a stack trace back to identify the file, removing the throw in the XH file. Should we use this approach? 
```
if(XH._desktopAPIRegistered){
    throw "`Desktop components imported into mobile app.  Please check your imports.`";
 }
 XH._mobileAPIRegistered = true;
```

**Comment by TomTirapani** [comment]
Ah, so it gives a stack trace back to the offending file? Smart! I like that approach :)

There's a ticket for the Chart JS import: https://github.com/xh/hoist-react/issues/3075

For the DockView, I'm surprised that's being included in mobile at all. I would also vote for moving it to dekstop, we have no plan (nor desire) for a mobile implementation.

File: desktop/register.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {XH} from '@xh/hoist/core';
+
+if(XH._mobileAPIRegistered){
```
Comment (lbwexler) [inline desktop/register.js:3]: fine, but we should use `XH.exception()` to create this exception for good measure.

Comment (lbwexler) [inline desktop/register.js:3]: add space after `if`

**Comment by lbwexler** [comment]
I also agree that, would be nice for the register import to be first in the file, but also know that there is a desire to let the imports be managed by the IntelliJ.

Would a line break help?

**Comment by Ryanseanlee** [comment]
Anselm and I also talked about having the import first and we found that the line break does not help as it just organizes it anyways. So, we decided to just let IntelliJ manage the imports.

**Review by lbwexler — APPROVED** [review]

**Comment by lbwexler** [comment]
I addressed the problem with ContextMenu to provide a dynamic implementation, and am reviewing that with Ryan.
This just extends a pattern we are already doing with a number of hoist implementations, no need for any API changes, or introducing new APIs.

Ryan will move the DockViewModel and we should be good to go!

**Comment by Ryanseanlee** [comment]
https://github.com/xh/toolbox/pull/594 This is the toolbox update to have the correct import location for the DockView.

**Pull request merged**